tests/functional/__init__.py

```diff
@@ -10,3 +10,31 @@
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
+import os
+import uuid
+
+_ORIGINAL = os.environ.copy()
+# These are environment variables that allow users to control
+# the location of config files used by botocore.
+_CONFIG_FILE_ENV_VARS = [
+    'AWS_CONFIG_FILE',
+    'AWS_SHARED_CREDENTIALS_FILE',
+    'BOTO_CONFIG',
+]
+_CREDENTIAL_ENV_VARS = [
+    'AWS_ACCESS_KEY_ID',
+    'AWS_SECRET_ACCESS_KEY',
+    'AWS_SESSION_TOKEN',
+]
+
+
+def setup_package():
+    random_file = str(uuid.uuid4())
+    for varname in _CONFIG_FILE_ENV_VARS:
+        os.environ[varname] = random_file
```

Comment:
Makes sense, and sounds reasonable to me. I guess my only suggestion is a comment to mention that, so that 6 months from now we don't come back and try to figure out why this code seemingly duplicates ``BaseSessionTest``.